src/main/src/test/java/org/geoserver/filters/BufferedRequestStreamTest.java

```diff
@@ -32,6 +34,32 @@ public void testReadLine() throws Exception {
         int len = 1024;
         int amountRead = myBRS.readLine(b, off, len);
         String s = new String(b, 0, amountRead);
-        assertEquals(s, myTestString);
+        assertEquals(myTestString, s);
+    }
+
+    @Test
+    public void closeClose() throws IOException {
+        byte[] b = new byte[1024];
+        int off = 0;
+        int len = 1024;
+        int amountRead = myBRS.readLine(b, off, len);
+        String s = new String(b, 0, amountRead);
+        assertEquals(myTestString, s);
+        myBRS.reset();
+
+        amountRead = myBRS.readLine(b, off, len);
+        String s2 = new String(b, 0, amountRead);
+        assertEquals(myTestString, s2);
+
+        myBRS.close();
+
+        try {
+            amountRead = myBRS.readLine(b, off, len);
+            String s3 = new String(b, 0, amountRead);
+            assertEquals(myTestString, s3);
+            fail("Buffered Request Stream should already be closed");
+        } catch (IOException closed) {
+            // expected
```

Comment:
You aren't really testing that the exception is triggered. Maybe you should use the JUnit `@Test(expected = IOException.class)` method annotation instead of just `@Test`

Reply:
Okay; I will check the specific IOException message is present.